interface-definitions/dhcp-server.xml.in

```diff
@@ -129,7 +129,7 @@
                     <properties>
                       <help>Bootstrap file name</help>
                       <constraint>
-                        <regex>[-_a-zA-Z0-9./]+</regex>
+                        <regex>[-_a-zA-Z0-9.:/~#@!$&*+,;=~%\\\\]+</regex>
```

Comment:
Why not use something like: `<regex>[[:ascii:]]{0,256}</regex>` to allow all ASCII characters? Makes it easier to read.

Reply:
As long as VyOS supports that then this `<regex>[[:ascii:]]{0,256}</regex>` a better option since the source code of ISC DHCP defines that option as type "t" meaning just that:

https://github.com/isc-projects/dhcp/blob/master/common/tables.c#L165

```
{ "bootfile-name", "t",			&dhcp_universe,  67, 1 },
```

```
/* DHCP Option names, formats and codes, from RFC1533.
   Format codes:
...
   t - ASCII text
```

RFC1533 has been superseeded by RFC2132 according to IANA:

https://www.iana.org/assignments/bootp-dhcp-parameters/bootp-dhcp-parameters.xhtml

https://www.rfc-editor.org/rfc/rfc2132.html

And in RFC2132:

```
9.5 Bootfile name

   This option is used to identify a bootfile when the 'file' field in
   the DHCP header has been used for DHCP options.

   The code for this option is 67, and its minimum length is 1.

       Code  Len   Bootfile name
      +-----+-----+-----+-----+-----+---
      | 67  |  n  |  c1 |  c2 |  c3 | ...
      +-----+-----+-----+-----+-----+---
```

Since the above says that length must be at least 1 shouldnt the regex in question be this instead?

```
<regex>[[:ascii:]]{1,256}</regex>
```

Reply:
Please use 1 as minimum length

Reply:
Thanks -- I agree that `[[:ascii:]]` would make the pattern easier to read / maintain. The original proposal was meant to align with the established conventions for this and other value constraints. If we're OK with using the POSIX expression here, I agree that's preferable.

In that case, should we consider adjusting the byte limits to a minimum of `1` as previously mentioned, and a maximum of `253`, to account for the maximum option length of `255` bytes, minus two for the option tag and length bytes?

This would result in the final pattern of `[[:ascii:]]{1,253}`. If this looks good, I'll go ahead and retest / resubmit.

Reply:
Please do so

Reply:
'[[:ascii:]]{1,253}' sounds good to me too.

As a spinoff there might be needed to add an additional task of verifying all other options available through the dhcp config in VyOS aswell.

That is '[[:ascii:]]{1,253}' should be used for all variables (dhcp options) defined as type "t" in https://github.com/isc-projects/dhcp/blob/master/common/tables.c#L43-L301